### PR TITLE
feature: complete modular starter module

### PR DESCRIPTION
**Pull request by jaredfholgate**
## Overview/Summary

The PR adds an end to end example of using the new avm modules for ALZ.

## This PR fixes/adds/changes/removes

1. #29

### Breaking Changes

None

## Testing Evidence

The starter module has been run and generated a hierarchy with no errors:
image

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/alz-terraform-accelerator/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/alz-terraform-accelerator/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/alz-terraform-accelerator/tree/main)
- [x] Performed testing and provided evidence.
- [x] Updated relevant and associated documentation.

### CODE REVIEW

**Comment by luke-taylor** [comment]
Nice work here Jared. 

I'd be tempted to remove the template file variables though, it layers in more complexity. I think being more opinionated would be better, the idea behind the complete was to somewhat replicate the scalability of caf-enterprise-scale. 

Do you think we could just hard code certain variables directly in the modules in main.tf, like log_analytics_workspace_id etc. I don't think they necessarily need exposing in the config.yaml as 9 times out of 10 it will be the same, i.e. the output of the alz-management module.

**Comment by jaredfholgate** [comment]
> Nice work here Jared.
> 
> I'd be tempted to remove the template file variables though, it layers in more complexity. I think being more opinionated would be better, the idea behind the complete was to somewhat replicate the scalability of caf-enterprise-scale.
> 
> Do you think we could just hard code certain variables directly in the modules in main.tf, like log_analytics_workspace_id etc. I don't think they necessarily need exposing in the config.yaml as 9 times out of 10 it will be the same, i.e. the output of the alz-management module.

It is quite possible I am over-engineering it. I was just thinking ahead to VBD usage and keeping it DRY. But yes, the templating could be confusing if not well documented. I will have a think about how to simplify it. I think the log_analytics_workspace_id can be removed for sure. The subscription ids might be problematic, but sure there is a way to handle that.

**Comment by luke-taylor** [comment]
> > Nice work here Jared.
> > I'd be tempted to remove the template file variables though, it layers in more complexity. I think being more opinionated would be better, the idea behind the complete was to somewhat replicate the scalability of caf-enterprise-scale.
> > Do you think we could just hard code certain variables directly in the modules in main.tf, like log_analytics_workspace_id etc. I don't think they necessarily need exposing in the config.yaml as 9 times out of 10 it will be the same, i.e. the output of the alz-management module.
> 
> It is quite possible I am over-engineering it. I was just thinking ahead to VBD usage and keeping it DRY. But yes, the templating could be confusing if not well documented. I will have a think about how to simplify it. I think the log_analytics_workspace_id can be removed for sure. The subscription ids might be problematic, but sure there is a way to handle that.

Yeah for sure, and the folks delivering the VBDs may not necessarily be fully clued up on advanced features of terraform. Why may the subscription Ids be problematic?

File: templates/complete_modular/main.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
+module "management_resources" {
+  source  = "Azure/alz-management/azurerm"
+  version = "~> 0.1.0"
+
+  automation_account_name      = try(local.management.automation_account_name, "")
+  location                     = try(local.management.location, "")
+  log_analytics_workspace_name = try(local.management.log_analytics_workspace_name, "")
+  resource_group_name          = try(local.management.resource_group_name, "")
```
Comment (luke-taylor) [inline templates/complete_modular/main.tf:8]:
```suggestion
  resource_group_name          = try(local.management.resource_group_name, "")
    providers = {
    azurerm = azurerm.management
  }
```
Perhaps we could explicitly declare the provider?

**Comment by matt-FFFFFF** [comment]
@jaredfholgate see this branch for some progress:

https://github.com/Azure/terraform-azurerm-avm-ptn-alz/tree/experiment/yamlinput/examples/dynamic-input

**Review by matt-FFFFFF — APPROVED** [review]
lgtm

**Pull request merged**